catalog/app/containers/Admin/BucketConfigSelection.generated.ts

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```

Comment:
correct

> The source of truth for these generated types somewhere in GraphQL?

for queries / mutations `path/to/file.graphql` becomes `path/to/file.generated.ts`, for schema it's a little different -- the schema file lives in the `shared` folder and generated code for it lives in `app/model/graphql`

Reply:
but you should read a couple of them anyways to understand what code (types and data) gets generated from graphql documents

Reply:
@nl0 
if these files are generated automatically do we really need to put them in repo?

Reply:
well, that's a good question and i don't have a good answer.
TLDR: probably we do for now, but ideally we shouldnt.

i hoped that there exists some tool that would do just that -- generate ts code from graphql queries without writing the actual files to disc (keeping them virtual), but unfortunately typescript doesnt make it easy to load custom file types and transform them into ts in memory (there is a candidate for this -- https://github.com/firede/ts-transform-graphql-tag, but it's abandoned and doesnt work with the current versions of the libraries).
we could use a webpack loader, but ts autocomplete wont work with this approach.
also, we could generate the files on the fly every time development server is started (or as a build step in case of production builds), which may be ok, actually.
anyways, i plan to iterate on all this code generation logic soonish to reduce the maintenance burden as much as i can.

Reply:
> also, we could generate the files on the fly every time development server is started (or as a build step in case of production builds), which may be ok, actually.

That sounds reasonable.

> anyways, i plan to iterate on all this code generation logic soonish to reduce the maintenance burden as much as i can.

:+1: